src/steamship_langchain/vectorstores/steamship_vector_store.py

```diff
@@ -0,0 +1,109 @@
+import uuid
+from itertools import zip_longest
+from typing import Any, Iterable, List, Optional
+
+from langchain.docstore.document import Document
+from langchain.vectorstores import VectorStore
+from steamship import Steamship, Tag
+
+
+class SteamshipVectorStore(VectorStore):
+    """Wrapper around Steamships vector database.
+
+    Example:
+        .. code-block:: python
+
+            from steamship_langchain import SteamshipVectorStore
+            faiss = SteamshipVectorStore(embedding_function, index, docstore)
+
+    """
+
+    def __init__(
+        self,
+        client: Steamship,
+        embedding: str,
+        index_name: Optional[str] = None,
+    ):
+        """Initialize with necessary components."""
+
+        self.client = client
+        self.index_name = index_name or uuid.uuid4().hex
+
+        self.index = client.use_plugin(
+            plugin_handle="embedding-index",
+            instance_handle=self.index_name,
+            config={
+                "embedder": {
+                    "plugin_handle": "openai-embedder-test",
+                    "instance_handle": self.index_name,
+                    "fetch_if_exists": True,
+                    "config": {
+                        "model": embedding,
+                        "dimensionality": 1536
+                    },
+                }
+            },
+            fetch_if_exists=False,
+        )
+
+    def add_texts(self, texts: Iterable[str], metadatas: Optional[List[dict]] = None) -> None:
+        """Run more texts through the embeddings and add to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+        """
+        items = [
+            Tag(client=self.client, text=text, value=metadata)
+            for i, (text, metadata) in enumerate(zip_longest(texts, metadatas or []))
+        ]
+
+        self.index.insert(items)
+
+    def similarity_search(self, query: str, k: int = 4, **kwargs: Any) -> List[Document]:
+        search_results = self.index.search(query, k=k)
+        search_results.wait()
```

Comment:
i don't think it matters too much, since results should return within 3m, but i'm wondering what guideline we want to establish for wrapping / not wrapping this in `try`.

Reply:
You mean when the .wait() times out? Good question, maybe in the future we expose the timeout in the vector store interface? That does deviate from LangChain's interface though which makes it less ideal.